functions/usage/index.js

```diff
@@ -57,30 +57,41 @@ function _getValue(data, key, default_value = '') {
     return value;
 }
 
+/**
+ * @param {string} org
+ * @param {string} userIP
+ */
 function getGeoLocationInfo(org, userIP) {
-    return new Promise(function (resolve, reject) {
-        try {
-            geoip2.init('geo/GeoLite2-City.mmdb');
-            geoip2.lookup(userIP, function (error, result) {
-                if (error) {
-                    console.log('GEOIP Error: %s', error);
-                    resolve({});
-                } else if (result) {
-                    console.log('GEOIP result: ' + JSON.stringify(result));
-                    result = getGeoIpInfo(result);
-                    result['org'] = org;
-                    console.log('GEOIP slim result: ' + JSON.stringify(result));
-                    resolve(result);
-                } else {
-                    reject(Error('say what??'));
-                }
-            });
-        } catch (err) {
-            console.log(`failed to retrieve geo ip info: ${err.message}`);
-            resolve({});
-        }
-    });
+    return geoip2.Reader.open('geo/GeoLite2-City.mmdb')
+        .then((reader) => {
+            const geoipResult = reader.city(userIP);
+            const subdivision = geoipResult.subdivisions?.[0].names.en ?? '';
+            const country = geoipResult.country?.names.en ?? '';
+            const city = geoipResult.city?.names.en ?? '';
+
+            return {
+                city,
+                continent: geoipResult.continent?.names.en ?? '',
+                country,
+                location: [city, subdivision, country]
+                    .filter((s) => s.length > 0)
+                    .join(', '),
+                org,
+                subdivision,
+                timezone: geoipResult.location?.timeZone,
+            };
+        })
+        .catch((error) => {
+            console.log('GEOIP Error: %s', error);
+
+            if (error?.name === 'AddressNotFoundError') {
+                return Promise.reject(error);
+            }
+
+            return {};
+        });
 }
+exports.getGeoLocationInfo = getGeoLocationInfo;
 
 function getGeoIpInfo(geoIpInfo) {
```

Comment:
I've just noticed that I need to take care of this function now.

Also, I see that `uptime` uses the same functions. I'll try to extract the implementation so it's not duplicated

Reply:
I have been trying to come up with some reasonable setup that would extract the common parts, but IMO then the complexity would be much higher, and I don't think that it's worth my time if we are not working on this repository :slightly_smiling_face: Thus, I'll just remove this unused function. When making the changes for the `uptime` function, I will just copy over the same implementation for `getGeoLocationInfo` that is done in this PR.

I am aware it will result in duplication of code, but IMO we can live with it with a repo that is hardly changed, and I there are items of higher priority in other repos :slightly_smiling_face: